### ISSUE TITLE
Target at least SDK 26

### ISSUE THREAD

**Issue by rubenwardy** [minetest/minetest#7813]
[Play store requires apps to target SDK 26](https://developer.android.com/distribute/best-practices/develop/target-sdk) at least. We currently target 14.

https://github.com/minetest/minetest/blob/master/build/android/build.gradle#L30

**Comment by stujones11** [comment]
AFAIK this only applies to new apps.

**Comment by rubenwardy** [comment]
This applies to all updates to the play store

**Comment by stujones11** [comment]
> This applies to all updates to the play store

I guess you need to release 5.0.0 before November then xD

Seriously though, it's not a big deal to change this when the time does come :)

**Comment by stujones11** [comment]
This turns out to be less trivial than I had expected. I was eventually able to make it build with ndk-18 by setting a minimum api of 16. However, when I launch the app, the loading bar just keeps repeating itself in an infinite cycle.

It seems we need some updates to the java code, of which I have no clue or desire to learn.

Feature branch here: https://github.com/stujones11/minetest/tree/android-ndk-18

**Comment by MoNTE48** [comment]
In api 16 you need request permission https://github.com/MultiCraftProject/MultiCraft/blob/master/build/android/src/main/java/mobi/MultiCraft/MainActivity.java#L270
@stujones11 if you are not a java programmer, you should not touch this topic. In the newer APIs, Google has made many changes that are required to be respected.

**Comment by stujones11** [comment]
> if you are not a java programmer, you should not touch this topic.

You are right, it seems a bit more complex than I had first imagined. The branch linked above has the makefile changes to build with the latest ndk-18, if that is any help.

**Comment by MoNTE48** [comment]
@stujones11 send me, please, your apk, I do not have a linux machine right now. (I need minetest libs for testing) I'll do it on the weekend

**Comment by stujones11** [comment]
Here is an apk built from my [android-ndk-18](https://github.com/stujones11/minetest/tree/android-ndk-18) branch.

https://nofile.io/f/2XMFtVEGCbY/Minetest-debug.apk

Since I have not been able to launch the main activity, I cannot guarantee these libs even work. There are, however, some known to be working apks linked in the gui scaling PR.

**Comment by MoNTE48** [comment]
File Not found

**Comment by stujones11** [comment]
> File Not found

It seem nofile.io is misbehaving at the moment, I have updated the link to a different file host, sorry if there as ads.

https://ufile.io/zr1qn

**Comment by rubenwardy** [comment]
You don't necessarily need Java experience to do that, there are tutorials available. Languages are basically all the same anyway

**Comment by MoNTE48** [comment]
https://nofile.io/f/dHhkTWXHptp/Minetest-debug.apk
can someone check it out?

**Comment by stujones11** [comment]
File Not found

**Comment by MoNTE48** [comment]
Sorry. https://ufile.io/y1yq2

**Comment by stujones11** [comment]
Crashes immediately on start-up. I am seeing lots of errors like this in the logcat
```
10-26 19:04:26.005 16911 16938 E MinetestAssetCopy: Failed to open asset file "/data/user/0/net.minetest.minetest/files/Minetest/games/minetest_game/settingtypes.txt" for size check
10-26 19:04:26.006 16911 16938 E MinetestAssetCopy: Failed to open asset file "/data/user/0/net.minetest.minetest/files/Minetest/games/minetest_game/README.txt" for size check
10-26 19:04:26.008 16911 16938 E MinetestAssetCopy: Failed to open asset file "/data/user/0/net.minetest.minetest/files/Minetest/games/minetest_game/mods/xpanes/license.txt" for size check
10-26 19:04:26.009 16911 16938 E MinetestAssetCopy: Failed to open asset file "/data/user/0/net.minetest.minetest/files/Minetest/games/minetest_game/mods/xpanes/init.lua" for size check
10-26 19:04:26.010 16911 16938 E MinetestAssetCopy: Failed to open asset file "/data/user/0
```

**Comment by MoNTE48** [comment]
Something with the new Android Studio 3.2.1, I'm exploring this. I have a successful launch with every second build. In vain I installed the update ...

**Comment by stujones11** [comment]
That update didn't work for me either, I have to cancel that option to make it build.

**Comment by MoNTE48** [comment]
@stujones11 https://ufile.io/78kbe

**Comment by stujones11** [comment]
WOW! That works very well!
![screen](https://user-images.githubusercontent.com/3710749/47591757-a867c800-d968-11e8-9f8e-b0127bab8134.png)
However, is the 'allow' access to camera and pictures directories dialog absolutely required? I did have it working without that using hard-coded paths. Maybe this you needed this for MultiCraft?

**Comment by MoNTE48** [comment]
Good :)
Starting with Android 6.0, android.permission.WRITE_EXTERNAL_STORAGE should be requested this way. This allows the application to access / sdcard. No other way. Now I will publish the source code. I think that someone will be able to suggest some improvements. I also remove from Android 2 styles and add Material Design styles.

**Comment by MoNTE48** [comment]
https://github.com/MoNTE48/minetest/commit/ce26c246baf6f715c2146b516d744f60f3802574
Note: 
1) WIP! 
2) I did not make a change in MinetestTextEntry and MinetestAssetCopy. But I did the automatic code formatting + import optimization in Android Studio. Therefore, it is not placed much differently, according to the standards.
3) I'm not a programmer, so I will be grateful if an experienced java dev checks the code.

**Comment by stujones11** [comment]
> This allows the application to access / sdcard

From what I have found, we don't actually need access to the storage directly if we use the proper user data path. I will investigate this some more.

> I think that someone will be able to suggest some improvements.

I am sure they will, everyone's a critic :) From my point of view, the install was fast, clean and more modern looking. Thank you for sharing this with us.

**Comment by MoNTE48** [comment]
Of course, the right decision to transfer files to the /data folder. But it will break applications such as Minetest Mods by @rubenwardy.
Also, if we want to transfer worlds to the worlds folder from sdcard, we need to get this permission to read the contents of the sdcard. New api :)
Therefore, you must have this memory access request to fulfill Google’s requirements. Or you lose the user worlds.

**Comment by stujones11** [comment]
@MoNTE48 I have tested your changes again using a clean build environment and everything seems to be fine on my Android 7 tablet. However, testing on my new Android 8 phone (Honor 9 lite) I get this when trying to access the virtual keyboard.
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: net.minetest.minetest, PID: 9167
    java.lang.RuntimeException: Unable to start activity ComponentInfo{net.minetest.minetest/net.minetest.minetest.MinetestTextEntry}: java.lang.IllegalStateException: Only fullscreen opaque activities can request orientation
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3194)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3302)
        at android.app.ActivityThread.-wrap12(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1891)
        at android.os.Handler.dispatchMessage(Handler.java:108)
        at android.os.Looper.loop(Looper.java:166)
        at android.app.ActivityThread.main(ActivityThread.java:7425)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
     Caused by: java.lang.IllegalStateException: Only fullscreen opaque activities can request orientation
        at android.app.Activity.onCreate(Activity.java:1081)
        at net.minetest.minetest.MinetestTextEntry.onCreate(MinetestTextEntry.java:23)
        at android.app.Activity.performCreate(Activity.java:7372)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1218)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3147)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3302) 
        at android.app.ActivityThread.-wrap12(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1891) 
        at android.os.Handler.dispatchMessage(Handler.java:108) 
        at android.os.Looper.loop(Looper.java:166) 
        at android.app.ActivityThread.main(ActivityThread.java:7425) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921) 
```
Otherwise it is playing quite nicely on this phone. I am really very happy with it for the price :)

![screen](https://user-images.githubusercontent.com/3710749/47608189-c9402400-da21-11e8-981d-a06affd6805b.png)

**Comment by stujones11** [comment]
@MoNTE48 Thank you again, that worked like a charm :)

**Comment by MoNTE48** [comment]
I think yes. But I would like to do some more small things. I will come back to this on Monday

**Comment by paramat** [comment]
Closed by #7820
